### PR TITLE
Modifying comments tests

### PR DESCRIPTION
**Pull request by SergioBertolinSG**
Refactoring and preparation for adding more tests related with comments.

### CODE REVIEW

**Comment by PVince81** [comment]
Had a quick glance, looks good 👍

**Comment by mrow4a** [comment]
Awesome, lets make CommentsApp great again!

**Comment by SergioBertolinSG** [comment]
This step is commented because it is failing:

https://github.com/owncloud/core/pull/27446/files#diff-6692878a78a86bbf0b3daf9e5e6522deR104

Raised here: https://github.com/owncloud/core/issues/27455

**Comment by PVince81** [comment]
```
17:58:15 --- FeatureContext has missing steps. Define them with these snippets:
17:58:15 
17:58:15     /**
17:58:15      * @Given As :arg1 sending :arg2 to :arg3 with
17:58:15      */
17:58:15     public function asSendingToWith($arg1, $arg2, $arg3, TableNode $table)
17:58:15     {
17:58:15         throw new PendingException();
17:58:15     }
```

**Comment by SergioBertolinSG** [comment]
Apparently tags tests depended on a comments function. I have moved it there. 
Tags will require the same changes as comments. But not to be done in this PR.

File: tests/integration/features/bootstrap/Comments.php [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+								  "PROPPATCH",
+								  $commentsPath,
+								  [],
+								  null,
+								  "uploads",
+								  '<?xml version="1.0"?>
+									<d:propertyupdate  xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns">
+										<d:set>
+											<d:prop>
+												<oc:message>'. $content .'</oc:message>
```
Comment (PVince81) [inline tests/integration/features/bootstrap/Comments.php]: would be safer to XML-encode this string to avoid injection.
Fortunately this is just test code

**Comment by PVince81** [comment]
Fix the XML encode issue and this is good to go 👍

File: tests/integration/features/bootstrap/Comments.php (hunk 1 above)
Comment (SergioBertolinSG) [inline tests/integration/features/bootstrap/Comments.php]: Do you mean escape? Is htmlspecialchars fine for this?

**Comment by SergioBertolinSG** [comment]
The property {http://owncloud.org/ns}comments-unread is not being included.

Jenkins failure:
```
Traza

And the single response should contain a property "{http://owncloud.org/ns}comments-unread" with value "0":

Cannot find property "{http://owncloud.org/ns}comments-unread" with "0"
```

Seems to be a bug introduced in https://github.com/owncloud/core/pull/27339 

cc @mrow4a

**Comment by mrow4a** [comment]
@SergioBertolinSG Did you check this scenario "by hand" first?

**Comment by SergioBertolinSG** [comment]
I checked it locally and it returned only the other two properties.

**Comment by mrow4a** [comment]
Normal propfind works. Will look at it later, but this might not be working previously also, dont see a point in the code yet.

**Comment by mrow4a** [comment]
@SergioBertolinSG this should be separate PR to this branch, isnt it?

**Comment by SergioBertolinSG** [comment]
> @SergioBertolinSG this should be separate PR to this branch, isnt it?

Ok, that failing test the only thing stopping this Pr to be merged now.

**Comment by PVince81** [comment]
@mrow4a please fix on this branch. CI will tell whether it was actually fixed.
Thanks

**Comment by mrow4a** [comment]
It occured, that when targeting single file with PROPFIND, it was returning null instead of 0, and requirement is to return the property. In web, it does not do propfinds to single files for unread, but this fix was needed anyways.

**Comment by PVince81** [comment]
:+1:

**Pull request merged**